src/echo.js

```diff
@@ -23,12 +27,23 @@ function _echo(opts, messages) {
   // allow strings starting with '-', see issue #20
   messages = [].slice.call(arguments, opts ? 0 : 1);
 
-  if (messages[0] === '-e') {
-    // ignore -e
+  var option = ['-e', '-n', '-ne', '-en'].indexOf(messages[0]);
+  var output;
+
+  if (option >= 0) {
+    // ignore options
     messages.shift();
+    output = format.apply(null, messages);
+    if (option === 0) {
+      // add newline if -n is not passed
+      output += '\n';
+    }
+  } else {
+    output = format.apply(null, messages) + '\n';
   }
 
-  console.log.apply(console, messages);
```

Comment:
I don't think it matters, because `console.log` is just a wrapper around `process.stdout.write`. See [this](https://github.com/nodejs/node/blob/master/lib/console.js#L43).